src/applications/model/crazytaxi-streaming-server.h

```diff
@@ -0,0 +1,86 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef CRAZY_TAXI_STREAMING_SERVER_H
+#define CRAZY_TAXI_STREAMING_SERVER_H
+
+#include "ns3/application.h"
+#include "ns3/event-id.h"
+#include "ns3/ptr.h"
+#include "ns3/ipv4-address.h"
+#include "ns3/random-variable-stream.h"
+#include "ns3/gaming-traffic.h"
+#include "ns3/gaming-streaming-server.h"
+#include "ns3/callback.h"
+#include <functional>
+
+namespace ns3 {
+
+  class Socket;
+  class Packet;
+
```

Comment:
These are the Socket and Packet classes defined in NS-3. 
In all other ns3 modules, they used these classes without any description, which I'm not sure if it is standard or what.
Maybe because the details of these classes can be found in the definition of each class.

Reply:
Is this a forward declaration or not?

Reply:
Honestly, Idk why, but maybe because I'm using this class in the following, and I should declare the class like this. Otherwise, I'll receive an error.

Reply:
I think what @MatteoDrago  meant was to add documentation to the class `CrazytaxiStreamingServer`.
Regarding the forward declaration I don't see any reference to those two classes in the header file, so I would remove them. If needed, include them in the implementation file

Reply:
I removed it and it still works fine.
I saw in many NS3 codes, they declare these basic classes, not sure but maybe they do it just for code readability and maintainability and these kinds of stuff.

Reply:
As I pointed out, this is NOT for code readability but it has a specific purpose in C++. 
Please, [check this out](https://stackoverflow.com/questions/4757565/what-are-forward-declarations-in-c) as it helped me as well to understand when and how to use it.

Reply:
Also, my "I don't know if this is case" referred to the fact that I didn't check the code before commenting, and I wasn't sure if you needed forward declaration in your code :)